### DISCUSSION TITLE
Problems while building maya-usd with MaterialX support

### DISCUSSION THREAD

**Discussion by Vsevolod1983** [Autodesk/maya-usd#1594] [Q&A] [answered]
Hello !
I'm trying to build maya-usd with MaterialX support in order to try it with hdRPR render delegate which supports MaterialX(now its 1.37 only, but we are going to get support for 1.38 as well)

I use this help file: https://github.com/Autodesk/maya-usd/blob/dev/doc/MaterialX.md

I have several issues/questions:
1. What does it mean Maya PR126 ? Looks like its not 126th PR in maya-usd repository. Is it PR for Maya 2022 itself or this is something else? Where I can find this pr126 ? Could you please explain in more details?
2. I get error that Maya 2022 I'm pointing to does not have LightApi version 2 (it has light api version 1 only). Not sure how to fix that. Is it related somehow to Maya PR126 I mentioned in point 1?

3.  Also, in forked autodesk MaterialX library function prototype was changed: MX_CORE_API vector<NodePtr> getShaderNodes. So it used to return unordered_set but now it returns vector.
It affects MaterialXFilter.cpp file in pixar usd on line 137.
However it looks like its easy to fix with:
        std::vector<mx::NodePtr> shaderNodes = mx::getShaderNodes(node, mx::SURFACE_SHADER_TYPE_STRING);
        std::unordered_set<mx::NodePtr> mxShaderNodes(shaderNodes.begin(), shaderNodes.end());
So, probably worth to mention in documentation.

Thank you !

**Comment by kxl-adsk** [comment]
Thanks for testing, @Vsevolod1983 .

Re 1 - Maya PR126 is our beta release. You will need beta access to download it.
Re 2 - Maya 2022 Gold doesn't have the new API. The next update for Maya 2022 will have it available.
Re 3 - I don't believe I run into this issue, but maybe something changed after. I'm not really answering how it happened, hoping @JGamache-autodesk can jump in here.

Cheers,
KxL

**Reply by Vsevolod1983** [reply]
Thank you for quick reply!

1. "Maya PR126 is our beta release" - is this beta release of Maya 2022 / Maya 2022 SDK ? We talk about maya itself not about maya-usd, right ?
2. Do you know when update for maya2022 will be available ?

**Reply by JGamache-autodesk** [reply]
Re 3: This one was fixed by https://github.com/autodesk-forks/MaterialX/commit/b140e028e61afbbfe6b85e4826f6ce4f46c2fae2 so you might want to re-fetch the tip of adsk_contrib/dev.

**Reply by kxl-adsk** [reply]
We only have beta releases for new versions of Maya, not updates. As for the date of the next update for Maya 2022 - it's not a great answer, but we are not allowed to provide it. I'm hoping you won't have to wait for very long.

**Comment by Vsevolod1983** [comment, marked as answer]
Maya 2022.1 update is launched today.
Guys, Does it include PR126 and light api version 2 which we need to build maya-usd with MaterialX support?

**Reply by kxl-adsk** [reply]
Yes, it does. That's the update I was saying you won't have to wait very long for ;)

**Reply by Vsevolod1983** [reply]
Thank you!
